shamil_latypov/main.cpp

```diff
@@ -143,60 +207,154 @@ class GraphPrinter {
+
+          added_vert_num++;
+        }
+      }
+      curr_vert_num++;
+    }
   }
 
-  GraphPrinter output_graph;
+  color_param++;
+  // Генерация зеленых ребер
+  for (const auto& vertex : graph.get_vertices()) {
+    int green_edge_chance = 10;
+    if (edges_chance(gen) < green_edge_chance) {
+      graph.add_edge(vertex.get_id(), vertex.get_id(), color_param);
+    }
+  }
+
+  color_param++;
+  // Генерация голубых ребер
+  for (const auto& depth : graph.get_vertices_ids()) {
+    for (auto vertex_id = depth.begin(); vertex_id != depth.end() - 1;
+         vertex_id++) {
+      int blue_edge_chance = 25;
+      if (edges_chance(gen) < blue_edge_chance) {
+        graph.add_edge(*vertex_id, *(vertex_id + 1), color_param);
+      }
+    }
+  }
+
+  color_param++;
+  // Генерация желтых ребер
+  double yellow_edge_percent = 100.0 / (double)(depth - 1);
+  int curr_depth = 0;
+  for (auto depth = (graph.get_vertices_ids()).begin();
+       depth != (graph.get_vertices_ids()).end() - 1; depth++) {
+    for (auto& vertex_id : *depth) {
+      // Вектор вершин уровнем глубже, которые не являются потомком j вершины
+      std::vector<VertexId> vert_ids_depth_deeper;
+      for (auto& vertex_id2 : *(depth + 1)) {
+        if (!graph.get_vertices()[vertex_id].is_connected(vertex_id2)) {
+          vert_ids_depth_deeper.push_back(vertex_id2);
+        }
+      }
+      // Проверка, что у нас есть "свободные" вершины для желтого ребра и рандом
+      // выпал удачно для генерации
+      if (vert_ids_depth_deeper.size() &&
+          (double)edges_chance(gen) <
+              yellow_edge_percent * (double)curr_depth) {
+        // Выбираем вершину уровнем глубже, которая не является
+        // потомком нашей вершины
+        std::uniform_int_distribution<> rand_vertex(
+            0, vert_ids_depth_deeper.size() - 1);
+        int second_vert_num = 0;
+        second_vert_num = rand_vertex(gen);
+
+        graph.add_edge(vertex_id, vert_ids_depth_deeper[second_vert_num],
+                       color_param);
+      }
+    }
+    curr_depth++;
+  }
+
+  color_param++;
+  // Генерация красных ребер
+  for (auto depth = graph.get_vertices_ids().begin();
+       depth != graph.get_vertices_ids().end() - 2; depth++) {
+    for (const auto& vertex_id : *depth) {
+      double red_edge_chance = 33;
+      if (edges_chance(gen) < red_edge_chance) {
+        // Выбираем рандомом вершину 2мя уровнями глубже
+        std::uniform_int_distribution<> rand_vertex(0, (depth + 2)->size() - 1);
+        int second_vert_num = rand_vertex(gen);
+
+        graph.add_edge(vertex_id, (*(depth + 2))[second_vert_num], color_param);
+      }
+    }
+  }
+
+  return graph;
+}
+
+int main() {
+  int depth = 0;
+  std::cout << "Input depth of graph:\n";
+  std::cin >> depth;
+  int new_vertices_num = 0;
+  std::cout << "Input number of vertecies from each vertex:\n";
+  std::cin >> new_vertices_num;
+
+  Graph graph;
+  graph = generate_graph(depth, new_vertices_num);
```

Comment:
Сам себя перехитрил))) Это такой партизанский ход, чтобы враги не поняли, что ты задумал)
1. Зачем в 2 строки разбил.
2. Константу забыл
```suggestion
const auto graph = generate_graph(depth, new_vertices_num);
```